internal/preferences.go

```diff
@@ -0,0 +1,140 @@
+package internal
+
+import (
+	"sync"
+
+	"fyne.io/fyne"
+)
+
+// InMemoryPreferences provides an implementation of the fyne.Preferences API that is stored in memory.
+type InMemoryPreferences struct {
+	values   map[string]interface{}
+	lock     sync.RWMutex
+	OnChange func()
+}
+
+// Declare conformity with Preferences interface
+var _ fyne.Preferences = (*InMemoryPreferences)(nil)
+
+func (p *InMemoryPreferences) set(key string, value interface{}) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	p.values[key] = value
+	p.fireChange()
+}
+
+func (p *InMemoryPreferences) get(key string) (interface{}, bool) {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	v, err := p.values[key]
+	return v, err
+}
+
+// Values provides access to the underlying value map - for internal use only...
```

Comment:
If intented for internal use only should we make it private ?

Reply:
We can't the application based properties implementation needs to serialise the content. There might be a better way to do it but I can't figure it out.
Also this is all in the internal package so it shouldn't matter to our API...